pkg/imagestream/imagestream.go

```diff
@@ -189,6 +190,62 @@ func (is *imageStream) GetImageOfImageStream(ctx context.Context, dgst digest.Di
 	return &img, nil
 }
 
+func (is *imageStream) ResolveUpstreamRef(ctx context.Context, dgst digest.Digest) (reference.DockerImageReference, rerrors.Error) {
+	layers, rErr := is.imageStreamGetter.layers()
+	if rErr != nil {
+		return reference.DockerImageReference{}, rerrors.NewError(
+			ErrImageStreamUnknownErrorCode,
+			fmt.Sprintf("ResolveUpstreamRef: failed to get layers for image stream %s", is.Reference()),
+			rErr,
+		)
+	}
+
+	parent := ""
+	for image, ibr := range layers.Images {
+		found := false
+		for _, m := range ibr.Manifests {
+			if m == dgst.String() {
+				found = true
+				break
+			}
+		}
+		if found {
+			parent = image
+			break
+		}
+	}
+	if parent == "" {
+		return reference.DockerImageReference{}, rerrors.NewError(
+			ErrImageStreamImageNotFoundCode,
+			fmt.Sprintf("ResolveUpstreamRef: unable to find parent for image %s in image stream %s", dgst.String(), is.Reference()),
+			nil,
```

Comment:
This can't be nil, you can give it a `kerrors.NewNotFound(imageapiv1.Resource("imagestreamimage"), dgst.String())`. It'll weirdly panic and recover otherwise - at least that's what I've seen on unit tests.

Reply:
This is what I've seen in the tests:
```
    logrus.go:29: time="2022-12-23T14:56:11+01:00" level=error msg="manifestService.Get: unable to get image sha256:abd132 in imagestream user/app: %!v(PANIC=Error method: runtime error: invalid memory address or nil pointer dereference)"
```
It was probably me causing it by refactoring the openshifit fake, but still.
(I temporarily stole ResolveUpstreamRef)